SocialEngine/docker-php-apache
# Fix 'AH01630 client denied by server configuration' error

### What is the problem / feature ?

After docker-library/php#251, overwriting `apache2.conf` is no longer enough. This broke our image with `AH01630 client denied by server configuration` error (docker-library/php#258), because we use a custom path (instead of `/var/www`) for serving files.
### How did it get fixed / implemented ?

Switched from using a custom `apache2.conf` to overwriting `000-default.conf` virtual host, and in cases where `/app` is a volume (such as local development), to use env vars for setting the user for running apache.
### How can someone test / see it ?
1. Build a local version of this image: `docker build --pull -t socialengine/php-apache:5.6 versions/5.6`
2. Update one of broken downstream docker images
3. See that things are now fixed, with no change in functionality (local dev works, non-volume shared running works)

**Note:** Most of the "meat" of this PR is in bb9104a
